src/frame/sdk/GroupApi.ts

```diff
@@ -86,17 +87,55 @@ export function get(groupID: string) {
     return GroupApiEndpoints.callGroupGetApi(groupID).map(formatDetailedGroupResponse);
 }
 
+/**
+ * Create the error strings for missing users
+ */
+function createErrorForInvalidUserList(userKeysResponse: UserKeyListResponseType, memberList: string[]) {
+    const existingUserIDs = userKeysResponse.result.map(({id}) => id);
+    return memberList.filter((userID) => !existingUserIDs.includes(userID));
+}
+
+/**
+ * Checkes that all members listed in member list are existing members
+ */
+function validateMemberList(memberList: string[]) {
+    return memberList !== undefined && memberList.length > 0
+        ? UserApiEndpoints.callUserKeyListApi(memberList).flatMap((userKeysResponse) => {
+              if (userKeysResponse.result.length !== memberList.length) {
+                  const missingUsersString = createErrorForInvalidUserList(userKeysResponse, memberList);
+                  return Future.reject(
+                      new SDKError(
+                          new Error(`Failed to create group due to unknown users in user list. Missing user IDs: [${missingUsersString}]`),
+                          ErrorCodes.GROUP_CREATE_WITH_MEMBERS_FAILURE
+                      )
+                  );
+              }
+              return Future.of(userKeysResponse.result.map((user) => ({id: user.id, masterPublicKey: user.userMasterPublicKey})));
+          })
+        : Future.of([]);
+}
+
 /**
  * Create a new group with the provided ID and options
  * @param {string}  groupID Client provided optional group ID
  * @param {string}  groupName Client provide optonal group name
  * @param {boolean} addAsMember Whether the group creator shoudl be added as a member upon create
  * @param {boolean} needsRotation Whether the group needs to rotate it private key
  */
-export function create(groupID: string, groupName: string, addAsMember: boolean, needsRotation: boolean) {
-    return GroupOperations.groupCreate(ApiState.userPublicKey(), ApiState.signingKeys(), addAsMember)
-        .flatMap(({encryptedGroupKey, groupPublicKey, transformKey}) =>
-            GroupApiEndpoints.callGroupCreateApi(groupID, groupPublicKey, encryptedGroupKey, needsRotation, groupName, transformKey)
+export function create(
+    groupID: string,
+    groupName: string,
+    addAsMember: boolean,
+    needsRotation: boolean,
+    memberList?: string[]
+): Future<SDKError, GroupDetailResponse> {
+    const creatorMemberKey = addAsMember ? [{id: ApiState.user().id, masterPublicKey: publicKeyToBase64(ApiState.userPublicKey())}] : [];
+    const memberListValidationResult = memberList !== undefined && memberList.length > 0 ? validateMemberList(memberList) : Future.of([]);
```

Comment:
The `validateMemberList` method is already checking if the provided list is undefined, so you could probably just leave that here and then instead of creating the `memberListValidationResult` variable, just immediately do 

```
return validateMemberList(memberList)
	.flatMap((memberKeys) => ...
```